starknet_py/utils/typed_data.py

```diff
@@ -0,0 +1,115 @@
+from dataclasses import dataclass
+from typing import Union, Dict, List
+
+from starkware.cairo.common.hash_state import compute_hash_on_elements
+from starkware.starknet.public.abi import get_selector_from_name
+
+from starknet_py.cairo.felt import encode_shortstring
+from starknet_py.utils.typing import TypedDict
+
+
+class StarkNetDomain(TypedDict):
+    """
+    TypedDict representing a StarkNetDomain object
+    """
+
+    name: str
+    version: str
+    chainId: Union[str, int]
+
+
+@dataclass(frozen=True)
+class Parameter:
+    """
+    Dataclass representing a Parameter object
+    """
+
+    name: str
+    type: str
+
+
+@dataclass(frozen=True)
+class TypedData:
+    """
+    Dataclass representing a TypedData object
+    """
+
+    types: Dict[str, List[Parameter]]
+    primary_type: str
+    domain: StarkNetDomain
+    message: dict
+
+    def _encode_value(self, type_: str, value: Union[int, str]) -> str:
+        if type_[-1] == "*":
+            return compute_hash_on_elements(
+                [self.struct_hash(type_[:-1], data) for data in value]
+            )
+        if type_ in self.types:
+            return self.struct_hash(type_, value)
+        return int(get_hex(value), 16)
+
+    def _encode_data(self, type_: str, data: dict) -> List[str]:
+        values = []
+        for param in self.types[type_]:
+            encoded_value = self._encode_value(param.type, data[param.name])
+            values.append(encoded_value)
+
+        return values
+
+    def _get_dependencies(self, type_: str) -> List[str]:
+        if type_ not in self.types:
+            # type_ is a primitive type, has no dependencies
+            return []
+
+        dependencies = set()
+
+        def collect_deps(type_: str) -> None:
+            for param in self.types[type_]:
+                # strip the pointer
+                fixed_type = param.type[:-1] if param.type[-1] == "*" else param.type
+                if fixed_type in self.types and fixed_type not in dependencies:
+                    dependencies.add(fixed_type)
+                    # recursive call
+                    collect_deps(fixed_type)
+
+        # collect dependencies into a set
+        collect_deps(type_)
+        return [type_, *list(dependencies)]
+
+    def _encode_type(self, type_: str) -> str:
+        [primary, *dependencies] = self._get_dependencies(type_)
```

Comment:
```suggestion
        primary, *dependencies = self._get_dependencies(type_)
```
I think brackets are not needed here.